include/seqan3/range/view/interleave.hpp

```diff
@@ -0,0 +1,352 @@
+             std::Same<reference_t<urng_t>, reference_t<inserted_rng_t>>
+    //!\endcond
+view_interleave(urng_t &&, size_t, inserted_rng_t&&)
+    -> view_interleave<decltype(view::all(std::declval<urng_t>())), decltype(view::all(std::declval<inserted_rng_t>()))>;
+
+// ============================================================================
+//  interleave_fn (adaptor definition)
+// ============================================================================
+
+/*!\brief View adaptor definition for view::interleave.
+ */
+class interleave_fn : public pipable_adaptor_base<interleave_fn>
+{
+private:
+    //!\brief Type of the CRTP-base.
+    using base_t = pipable_adaptor_base<interleave_fn>;
+
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
+
+private:
+    //!\brief Befriend the base class so it can call impl().
+    friend base_t;
+
+    /*!\brief       For ranges that are viewable, delegate to std::view::all.
+     * \returns     An instance of std::view::all.
+     */
+    template <std::ranges::RandomAccessRange urng_t, std::ranges::RandomAccessRange inserted_rng_t,
+              std::Integral size_type>
+        requires std::ranges::SizedRange<urng_t> &&
+                 std::ranges::SizedRange<inserted_rng_t>
+    static auto impl(urng_t && urange, size_type size, inserted_rng_t && i)
+    {
+        return view_interleave{urange, size, i};
+    }
+
+    /*!\brief       For ranges that are not views and not lvalue-references, call view_interleave's constructor.
+     * \returns     An instance of seqan3::detail::view_interleave.
+     */
+    template <std::ranges::ForwardRange urng_t, std::ranges::ForwardRange inserted_rng_t, std::Integral size_type>
+    static auto impl(urng_t && urange, size_type size, inserted_rng_t && i)
+    {
+        return ranges::view::split(urange, size) | ranges::view::join(i);
+    }
+};
+
+} // namespace seqan3::detail
+
+// ============================================================================
+//  view::interleave (adaptor instance definition)
+// ============================================================================
+
+namespace seqan3::view
+{
+
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief A view that interleaves a given range into another range at regular intervals.
+ * \tparam urng_t The type of the range being processed.
+ * \tparam inserted_rng_t The type of the range being inserted.
+ * \param[in] urange The range being processed.
+ * \param[in] inserted_range The range being inserted.
+ * \param[in] step_size A value of size_type which indicates the interval to insert the inserted_range.
+ * \returns A range with the second range inserted at regular intervals. See below for properties of said range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view can be used to insert one range into another range at regular intervals.
```

Comment:
add:

It behaves essentially like `| ranges::view::split(step_size) | ranges::view::join(inserted_range)` except that for input that models std::ranges::RandomAccessRange and std::ranges::SizedRange a more efficient data structure is returned (otherwise it returns exactly the above combination of views).